pythx/middleware/toolname.py

```diff
@@ -0,0 +1,21 @@
+import logging
+from pythx.middleware.base import BaseMiddleware
+
+
+LOGGER = logging.getLogger("ClientToolNameMiddleware")
+
+
+class ClientToolNameMiddleware(BaseMiddleware):
+    def __init__(self, name="pythx"):
+        LOGGER.debug("Initializing")
+        self.name = name
```

Comment:
Are we planning to add additional middleware or allow devs to build custom ones off of PythX?

Reply:
Developers can add their own middlewares, simply by adding them to the `APIHandler`, like
```python
from foo import MyMiddleware
from pythx.api.handler import APIHandler
mw = MyMiddleware()
handler = APIHandler(middlewares=[mw])
```
and then passing that to the `Client` class. I'm also hacking up a passthrough so you can do something like
```python
c = Client(middlewares=[mw])
```
to completely abstract from the API handler class.